integration-tests/src/tests/client/cold_storage.rs

```diff
@@ -338,3 +339,91 @@ fn test_cold_db_copy_with_height_skips() {
         }
     }
 }
+
+/// Producing 4 epochs of blocks with some transactions.
+/// Call copying full contents of cold columns to cold storage in batches of specified max_size.
+/// Currently only supports super small or super big batch.
+/// If batch_size = 0, check that every value was copied in a separate batch.
+/// If batch_size = usize::MAX, check that everything was copied in one batch.
+/// Most importantly, checking that everything from cold columns was indeed copied into cold storage.
+fn test_initial_copy_to_cold(batch_size: usize) {
+    init_test_logger();
+
+    let epoch_length = 5;
+    let max_height = epoch_length * 4;
+
+    let mut genesis = Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+
+    genesis.config.epoch_length = epoch_length;
+    let mut chain_genesis = ChainGenesis::test();
+    chain_genesis.epoch_length = epoch_length;
+    let mut env = TestEnv::builder(chain_genesis)
+        .runtime_adapters(create_nightshade_runtimes(&genesis, 1))
+        .build();
+
+    let store = create_test_node_storage_with_cold(DB_VERSION, DbKind::Archive);
+
+    let mut last_hash = *env.clients[0].chain.genesis().hash();
+
+    for h in 1..max_height {
+        let signer = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+        for i in 0..5 {
+            let tx = SignedTransaction::send_money(
+                h * 10 + i,
+                "test0".parse().unwrap(),
+                "test1".parse().unwrap(),
+                &signer,
+                1,
+                last_hash,
+            );
+            env.clients[0].process_tx(tx, false, false);
+        }
+
+        let block = env.clients[0].produce_block(h).unwrap().unwrap();
+        env.process_block(0, block.clone(), Provenance::PRODUCED);
+        last_hash = block.hash().clone();
+    }
+
+    copy_all_data_to_cold(
+        (*store.cold_db().unwrap()).clone(),
+        &env.clients[0].runtime_adapter.store(),
+        batch_size,
+    )
+    .unwrap();
+
+    for col in DBCol::iter() {
+        if col.is_cold() {
```

Comment:
nit: invert logic and `continue` early - the code will be less nested and easier to read